pipeline/BuildBaseImage/Jenkinsfile

```diff
@@ -0,0 +1,42 @@
+node{
+    timestamps{
+        deleteDir()
+        checkout(
+        [$class: 'GitSCM', branches: [[name: 'master']],
+        extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'on-build-config']],
+        userRemoteConfigs: [[url: 'https://github.com/RackHD/on-build-config']]])
+
+        def shareMethod = load("on-build-config/jobs/ShareMethod.groovy")
+        def label_name = "smoke_test"
+        lock(label:label_name,quantity:1){
+            node_name = shareMethod.occupyAvailableLockedResource(label_name, [])
+            node(node_name){
+                deleteDir()
+                checkout scm
+                stage("build base image"){
+                    withCredentials([
+                        usernamePassword(credentialsId: 'ff7ab8d2-e678-41ef-a46b-dd0e780030e1',
+                                         passwordVariable: 'SUDO_PASSWORD',
+                                         usernameVariable: 'SUDO_USER'),
+                        usernamePassword(credentialsId: 'rackhd-ci-docker-hub',
+                             passwordVariable: 'DOCKERHUB_PASS',
+                             usernameVariable: 'DOCKERHUB_USER')
+                    ]){
+                        sh """#!/bin/bash -x
+                        echo $SUDO_PASSWORD |sudo -S docker login -u $DOCKERHUB_USER -p $DOCKERHUB_PASS
```

Comment:
No, I don't upload it to dockerhub.
Because there will be many images under rackhd dockerhub if we upload all the base images to dockerhub. 
If users want to deploy image service within docker, they can use the image-service/Dockerfile directly.